astropy/coordinates/solar_system.py

```diff
@@ -0,0 +1,270 @@
+ephemerides from jplephem.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from collections import OrderedDict
+import numpy as np
+from .sky_coordinate import SkyCoord
+from ..utils.data import download_file
+from .. import units as u
+from ..constants import c as speed_of_light
+from .representation import CartesianRepresentation
+from .builtin_frames import GCRS, ICRS
+from .builtin_frames.utils import get_jd12, cartrepr_from_matmul
+from .. import _erfa
+
+__all__ = ["get_body", "get_moon", "get_body_barycentric", "SOLAR_SYSTEM_BODIES"]
+
+KERNEL = None
+
+"""
+each value in the BODIES dictionary a list of kernel pairs needed
+to find the barycentric position of that object from the JPL kernel.
+"""
+BODY_NAME_TO_KERNEL_SPEC = OrderedDict(
+                                      (('sun', [(0, 10)]),
+                                       ('mercury', [(0, 1), (1, 199)]),
+                                       ('venus', [(0, 2), (2, 299)]),
+                                       ('earth-moon-barycenter', [(0, 3)]),
+                                       ('earth',  [(0, 3), (3, 399)]),
+                                       ('moon', [(0, 3), (3, 301)]),
+                                       ('mars', [(0, 4)]),
+                                       ('jupiter', [(0, 5)]),
+                                       ('saturn', [(0, 6)]),
+                                       ('uranus', [(0, 7)]),
+                                       ('neptune', [(0, 8)]),
+                                       ('pluto', [(0, 9)]))
+                                      )
+SOLAR_SYSTEM_BODIES = tuple(BODY_NAME_TO_KERNEL_SPEC.keys())
+
+
+def _download_spk_file(url=('http://naif.jpl.nasa.gov/pub/naif/'
+                            'generic_kernels/spk/planets/de430.bsp'),
+                       show_progress=True):
+    """
+    Get the Satellite Planet Kernel (SPK) file from NASA JPL.
+
+    Download the file from the JPL webpage once and subsequently access a
+    cached copy. The default is the file de430.bsp.
+
+    This file is ~120 MB, and covers years ~1550-2650 CE [1]_.
+
+    .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
+    """
+    return download_file(url, cache=True, show_progress=show_progress)
+
+
+def _get_kernel(*args, **kwargs):
+    """
+    Try importing jplephem, download/retrieve from cache the Satellite Planet
+    Kernel.
+    """
+    global KERNEL
+
+    try:
+        from jplephem.spk import SPK
+    except ImportError:
+        raise ImportError("Solar system ephemeris calculations depend on"
+                          "the jplephem package")
+
+    if KERNEL is None:
+        KERNEL = SPK.open(_download_spk_file())
+    return KERNEL
+
+
+def get_body_barycentric(time, body):
+    """
+    Calculate the barycentric position of thesolar system body ``body``
```

Comment:
oops, just noticed a missing space here - feel free to fix it in a [ci skip] commit.  Or if the tests pass before you do so, can always just fix small things like this as part of the merge.